src/MooLite/core/inventory/Inventory.ts

```diff
@@ -87,23 +94,65 @@ export class Inventory {
         });
     }
 
-    updateCharacterDrink(actionTypeDrinkSlotsMap: Record<ActionTypeHrid, CharacterConsumable[]>) {
-        this._actionTypeDrinkSlotsMap = actionTypeDrinkSlotsMap;
+    updateCharacterDrink(actionTypeDrinkSlotsMap: Record<ActionTypeHrid, (CharacterConsumable | null)[]>) {
+        for (const actionType in actionTypeDrinkSlotsMap) {
+            const consumables = actionTypeDrinkSlotsMap[actionType];
+
+            if (this._characterDrinks[actionType] == null) {
+                this._characterDrinks[actionType] = consumables;
+            }
+
+            this._characterDrinks[actionType].forEach((storedConsumable, i) => {
+                if (storedConsumable === null) return;
+                const newConsumable = actionTypeDrinkSlotsMap[actionType][i];
+
+                // If stored consumable state is active and new state = inactive,
+                // dispatch the event. Checking for the difference stops the notification
+                // from being sent out every time updateCharacterDrink is called.
+                if (!newConsumable?.isActive && storedConsumable.isActive) {
+                    this._onConsumableDepleted.dispatch(storedConsumable);
+                }
+            });
+
+            // Save the new consumable slots state
+            this._characterDrinks[actionType] = actionTypeDrinkSlotsMap[actionType];
+        }
     }
 
     updateCharacterFood(actionTypeFoodSlotsMap: Record<ActionTypeHrid, CharacterConsumable[]>) {
```

Comment:
Could you add a `, notify: boolean = true` argument like in `Skills.ts` so that we can disable the notification from `InitCharacterInfo`?